### PR TITLE
Feature/employment history

### PR DESCRIPTION
**Pull request by L1ght-time**

### CODE REVIEW

File: src/components/shared/Card/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { createContext, FC } from 'react';
+
+import { TCardProps } from './Card.types';
+
+const CardContext = createContext(null);
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/Card.tsx:5]: For what?

File: src/pages/about/EmploymentHistory/EmploymentHistory.types.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+interface IPeriod {
+  start: string;
+  end: string;
+}
+
+interface IResponsibilities {
+  id: string;
+  item: string;
```
Comment (mykyta-lazurenko) [inline src/pages/about/EmploymentHistory/EmploymentHistory.types.ts:13]: Don't use 'item', 'data', etc.. 
Write 'company', 'profile', more understanding

File: src/pages/about/index.tsx [hunk 3]
```diff
@@ -2,29 +2,36 @@ import type { NextPage } from 'next';
 import { FormattedMessage } from 'react-intl';
 import process from 'process';
 
+import { IAboutData } from '@/pages/api/db/about';
+
+import { EmploymentHistory } from './EmploymentHistory';
+
 export async function getStaticProps() {
   try {
     const response = await fetch(`${process.env.API_HOST}/about`);
```
Comment (mykyta-lazurenko) [inline src/pages/about/index.tsx:11]: Use ky

File: src/pages/api/db/about.ts [hunk 4]
```diff
@@ -1,3 +1,27 @@
-export const about = {
+import { nanoid } from 'nanoid';
+
+import IEmploymentHistoryProps from '@/pages/about/EmploymentHistory';
+
+export interface IAboutData {
+  title: string;
+  employmentHistory: ReadonlyArray<IEmploymentHistoryProps>;
+}
```
Comment (mykyta-lazurenko) [inline src/pages/api/db/about.ts:8]: Indent

File: src/pages/api/db/about.ts [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+export const about: IAboutData = {
   title: 'ABOUT',
+  employmentHistory: [
+    {
+      id: nanoid(),
+      company: {
+        id: 0,
```
Comment (mykyta-lazurenko) [inline src/pages/api/db/about.ts:15]: Move image here

File: src/pages/about/EmploymentHistory/EmploymentHistory.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import { FC } from 'react';
+import Image from 'next/image';
+
+import civenty from '@/assets/civenty.jpeg';
```
Comment (mykyta-lazurenko) [inline src/pages/about/EmploymentHistory/EmploymentHistory.tsx:4]: Move that to the model, components do not need static if you can move in mapping

File: src/pages/about/EmploymentHistory/EmploymentHistory.types.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+interface ITechnologyStack {
+  id: string;
+  tag: string;
+}
+
+export default interface IEmploymentHistoryProps {
```
Comment (mykyta-lazurenko) [inline src/pages/about/EmploymentHistory/EmploymentHistory.types.ts:21]: Default? Its model won’t extend?

Comment (mykyta-lazurenko) [reply src/pages/about/EmploymentHistory/EmploymentHistory.types.ts:21]: I think better use naming export

File: src/pages/about/index.tsx [hunk 8]
```diff
@@ -2,29 +2,36 @@ import type { NextPage } from 'next';
...
-    const data = await response.json();
+    const { title, employmentHistory } = await response.json();
+
     return {
       props: {
-        title: data.title,
+        title,
```
Comment (mykyta-lazurenko) [inline src/pages/about/index.tsx:16]: Title - static, remove from model

File: src/pages/about/index.tsx [hunk 9]
```diff
@@ -2,29 +2,36 @@ import type { NextPage } from 'next';
...
     };
   } catch {
     return {
       props: {
         title: null,
+        employmentHistory: null,
       },
     };
   }
 }
```
Comment (mykyta-lazurenko) [inline src/pages/about/index.tsx:28]: Indent

File: src/pages/about/index.tsx [hunk 10]
```diff
@@ -2,29 +2,36 @@ import type { NextPage } from 'next';
...
-const About: NextPage = ({ title }: any) => {
-  if (!title) {
+const About: NextPage<IAboutData> = ({ title, employmentHistory }) => {
+  if (!title || !employmentHistory) {
     return null;
   }
 
-  console.log({ title });
+  console.log({ title, employmentHistory });
```
Comment (mykyta-lazurenko) [inline src/pages/about/index.tsx:34]: Logs

File: src/pages/about/index.tsx [hunk 11]
```diff
@@ -33,6 +40,13 @@ const About: NextPage = ({ title }: any) => {
       <h1 className="text-3xl font-bold underline text-red-700">
         <FormattedMessage id="title" />
       </h1>
+
+      <div>
+        <h2>Employment History:</h2>
```
Comment (mykyta-lazurenko) [inline src/pages/about/index.tsx:45]: Locale

File: src/pages/about/EmploymentHistory/EmploymentHistory.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+            <h2 className="card-title">{position}</h2>
+            <h2 className="card-title">{company.title}</h2>
+          </div>
+        </div>
+        <span>{`${period.start} - ${period.end}`}</span>
+
+        {responsibilities.map(({ id, item }) => (
+          <span key={id}>{item}</span>
+        ))}
+        <p>Skills:</p>
```
Comment (mykyta-lazurenko) [inline src/pages/about/EmploymentHistory/EmploymentHistory.tsx:32]: The same

**Review by mykyta-lazurenko — CHANGES_REQUESTED** [review]

File: src/components/shared/Card/Card.tsx (hunk 1 above)
Comment (L1ght-time) [reply src/components/shared/Card/Card.tsx:5]: May be for theme

File: src/pages/about/index.tsx (hunk 9 above)
Comment (L1ght-time) [reply src/pages/about/index.tsx:28]: lint doesn't low me do it.

File: src/components/shared/Card/Card.tsx (hunk 1 above)
Comment (mykyta-lazurenko) [reply src/components/shared/Card/Card.tsx:5]: I don't think that's necessary right now.

File: src/components/shared/Card/CardBadge/CardBadge.types.ts [hunk 13]
```diff
@@ -0,0 +1,3 @@
+import { PropsWithChildren } from 'react';
+
+export type TCardBadgeProps = PropsWithChildren<JSX.IntrinsicElements['span']>;
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/CardBadge/CardBadge.types.ts:3]:
```suggestion
export type TCardBadgeProps = JSX.IntrinsicElements['span'];
```

File: src/components/shared/Card/CardBody/CardBody.types.ts [hunk 14]
```diff
@@ -0,0 +1,3 @@
+import { PropsWithChildren } from 'react';
+
+export type TCardBodyProps = PropsWithChildren<JSX.IntrinsicElements['div']>;
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/CardBody/CardBody.types.ts:3]:
```suggestion
export type TCardBodyProps = JSX.IntrinsicElements['div'];
```

File: src/components/shared/Card/CardImage/CardImage.types.ts [hunk 15]
```diff
@@ -0,0 +1,3 @@
+import { ImageProps } from 'next/dist/client/image';
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/CardImage/CardImage.types.ts:1]:
```suggestion
import { ImageProps } from 'next/image';
```

File: src/components/shared/Card/CardButton/CardButton.types.ts [hunk 16]
```diff
@@ -0,0 +1,3 @@
+import { PropsWithChildren } from 'react';
+
+export type CardButtonProps = PropsWithChildren<JSX.IntrinsicElements['button']>;
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/CardButton/CardButton.types.ts:3]:
```suggestion
export type CardButtonProps = JSX.IntrinsicElements['button'];
```

File: src/components/shared/Card/CardText/CardText.types.ts [hunk 17]
```diff
@@ -0,0 +1,3 @@
+import { PropsWithChildren } from 'react';
+
+export type TCardTextProps = PropsWithChildren<JSX.IntrinsicElements['p']>;
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/CardText/CardText.types.ts:3]:
```suggestion
export type TCardTextProps = JSX.IntrinsicElements['p'];
```

File: src/components/shared/Card/CardWrapper/CardWrapper.tsx [hunk 18]
```diff
@@ -0,0 +1,5 @@
+import { FC } from 'react';
+
+import { TCardWrapperProps } from './CardWrapper.types';
+
+export const CardWrapper: FC<TCardWrapperProps> = ({ children, ...rest }) => <div {...rest}>{children}</div>;
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/CardWrapper/CardWrapper.tsx:5]: Useless component

File: src/components/shared/Card/index.ts [hunk 19]
```diff
@@ -0,0 +1,6 @@
+export { Card } from './Card';
+export { CardBody } from './CardBody';
+export { CardButton } from './CardButton';
+export { CardImage } from './CardImage';
+export { CardBadge } from './CardBadge';
+export { CardText } from './CardText';
```
Comment (mykyta-lazurenko) [inline src/components/shared/Card/index.ts:6]: Useless components

File: src/components/shared/Title/Title.tsx [hunk 20]
```diff
@@ -0,0 +1,12 @@
+import { FC } from 'react';
+
+import { TTitleProps } from './Title.types';
+
+export const Title: FC<TTitleProps> = ({ headingLevel, children, className, ...rest }) => {
+  const Heading = headingLevel;
+  return (
+    <Heading className={className} {...rest}>
```
Comment (mykyta-lazurenko) [inline src/components/shared/Title/Title.tsx:8]:
```suggestion
    <Heading {...rest}>
```

File: src/components/shared/Title/Title.types.ts [hunk 21]
```diff
@@ -0,0 +1,5 @@
+import { PropsWithChildren } from 'react';
+
+type THeading = { headingLevel: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' };
+
+export type TTitleProps = THeading & PropsWithChildren<JSX.IntrinsicElements[THeading['headingLevel']]>;
```
Comment (mykyta-lazurenko) [inline src/components/shared/Title/Title.types.ts:5]:
```suggestion
type THeading = { as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' };

export type TTitleProps = THeading & JSX.IntrinsicElements[THeading['as']];
```

File: src/components/shared/Title/Title.tsx [hunk 22]
```diff
@@ -0,0 +1,12 @@
+import { FC } from 'react';
+
+import { TTitleProps } from './Title.types';
+
+export const Title: FC<TTitleProps> = ({ headingLevel, children, className, ...rest }) => {
```
Comment (mykyta-lazurenko) [inline src/components/shared/Title/Title.tsx:5]: But I think it's a useless component

**Review by mykyta-lazurenko — CHANGES_REQUESTED** [review]

File: package.json [hunk 23]
```diff
@@ -10,20 +10,24 @@
...
     "@next/font": "13.1.6",
+    "@types/luxon": "^3.2.0",
     "@types/node": "18.13.0",
     "@types/react": "18.0.27",
     "@types/react-dom": "18.0.10",
     "clsx": "^1.2.1",
     "daisyui": "^2.50.1",
     "eslint": "8.33.0",
     "eslint-config-next": "13.1.6",
+    "ky": "^0.33.2",
```
Comment (mykyta-lazurenko) [inline package.json:21]: Remove

File: src/components/pages/About/About.tsx [hunk 24]
```diff
@@ -0,0 +1,32 @@
+import { NextPage } from 'next';
+import ky from 'ky';
+import { FormattedMessage } from 'react-intl';
+import useSWR from 'swr';
+
+import { IAboutData } from '@/pages/api/db/about';
+import { EmploymentHistory } from '@/components/pages/About/EmploymentHistory';
+
+// const fetcher = ky.get('/api/about').json;
+const fetcher = (...args: any[]) => fetch(...(args as [any, any])).then((res) => res.json());
```
Comment (mykyta-lazurenko) [inline src/components/pages/About/About.tsx:10]: Move to global constants

File: src/components/pages/About/About.tsx [hunk 25]
```diff
@@ -0,0 +1,32 @@
...
+
+export const About: NextPage = () => {
+  const { data: about } = useSWR<IAboutData>('/api/about', fetcher);
```
Comment (mykyta-lazurenko) [inline src/components/pages/About/About.tsx:13]: Better move to `useAbout` hook

File: src/components/pages/About/EmploymentHistory/helper/getDiffPeriod.ts [hunk 26]
```diff
@@ -0,0 +1,23 @@
...
+  if (monthsAmount < 12) {
+    return `${monthsAmount.toFixed(1)} months`;
+  }
+
+  if (months === 12) {
+    return `${(monthsAmount / 12).toFixed(0)} year`;
+  }
+
+  return `${monthsAmountMoreYear} years`;
+};
```
Comment (mykyta-lazurenko) [inline src/components/pages/About/EmploymentHistory/helper/getDiffPeriod.ts:23]: Localization?

File: src/components/pages/About/EmploymentHistory/helper/getDiffPeriod.ts [hunk 27]
```diff
@@ -0,0 +1,23 @@
+interface IDiffPeriodProps {
+  months: number;
+}
+
+export const getAmountOfPeriod = ({ months }: IDiffPeriodProps): string | null => {
```
Comment (mykyta-lazurenko) [inline src/components/pages/About/EmploymentHistory/helper/getDiffPeriod.ts:5]: Here needs only `months`, remove the object notation

**Review by mykyta-lazurenko — APPROVED** [review]

**Comment by L1ght-time** [comment]
I will fix the comments in the next PR.

**Pull request merged**